traitlets/traitlets.py

```diff
@@ -855,7 +857,7 @@ def set_metadata(self, key, value):
         warn("Deprecated in traitlets 4.1, " + msg, DeprecationWarning, stacklevel=2)
         self.metadata[key] = value
 
-    def tag(self, **metadata):
+    def tag(self, **metadata) -> "TraitType[G, S]":
```

Comment:
I think with the `typing_extension` package we can do this:
```suggestion
    def tag(self, **metadata) -> typing_extension.Self":
```